pdns/recursordist/settings/cxxsupport.cc

```diff
@@ -119,18 +122,172 @@ void pdns::settings::rec::oldStyleAllowFileToBridgeStruct(const std::string& fil
   }
 }
 
-static void mergeYamlSubFile(const std::string& configname, Recursorsettings& settings, Logr::log_t log)
+static void mergeYamlSubFile(const std::string& configname, Recursorsettings& settings, bool allowabsent, Logr::log_t log)
 {
-  SLOG(g_log << Logger::Notice << "Processing YAML settings from " << configname << endl,
-       log->info(Logr::Notice, "Processing YAML settings", "path", Logging::Loggable(configname)));
   auto file = ifstream(configname);
   if (!file.is_open()) {
+    if (allowabsent) {
+      return;
+    }
     throw runtime_error("Cannot open " + configname);
   }
+  SLOG(g_log << Logger::Notice << "Processing YAML settings from " << configname << endl,
+       log->info(Logr::Notice, "Processing YAML settings", "path", Logging::Loggable(configname)));
   auto data = string(std::istreambuf_iterator<char>(file), std::istreambuf_iterator<char>());
   pdns::rust::settings::rec::merge(settings, data);
 }
 
+static void possiblyConvertACLFile(const string& includeDir, const string& apiDir, const std::string& filename, Logr::log_t log)
+{
+  auto path = includeDir;
+  path.append("/").append(filename).append(".conf");
+  auto file = ifstream(path);
+  if (!file.is_open()) {
+    // Not an error, file just is not there
+    return;
+  }
+  rust::vec<rust::string> result;
+  std::string line;
+  while (getline(file, line)) {
+    auto pos = line.find('#');
+    if (pos != string::npos) {
+      line.resize(pos);
+    }
+    boost::trim(line);
+    if (line.empty()) {
+      continue;
+    }
+    auto plusis = line.find("+=");
+    if (plusis != string::npos) {
+      auto val = line.substr(plusis + 2);
+      boost::trim(val);
+      result.emplace_back(val);
+    }
+  }
+
+  rust::string key = "allow_from";
+  rust::string filekey = "allow_from_file";
+  if (filename == "allow-notify-from") {
+    key = "allow_notify_from";
+    filekey = "allow_notify_from_file";
+  }
+  const auto yaml = pdns::rust::settings::rec::allow_from_to_yaml_string_incoming(key, filekey, result);
+
+  string yamlfilename = apiDir;
+  yamlfilename.append("/").append(filename).append(".yml");
+  string tmpfilename = yamlfilename + ".tmp";
+  ofstream ofconf(tmpfilename);
+  if (!ofconf) {
+    int err = errno;
+    log->error(Logr::Error, err, "Cannot open for file for writing YAML", "to", Logging::Loggable(tmpfilename));
+    throw runtime_error("YAML Conversion");
+  }
+  ofconf << "# Generated by pdns-recursor REST API, DO NOT EDIT" << endl;
+  ofconf << yaml << endl;
+  ofconf.close();
+  if (rename(path.c_str(), (path + ".converted").c_str()) != 0) {
```

Comment:
We might want to look at `rdstate()` (or `bad()`) after calling `close()` and before renaming?